examples/platform/silabs/FreeRTOSRuntimeStats.c

```diff
@@ -183,30 +254,51 @@ uint32_t ulGetAllTaskInfo(TaskInfo * taskInfoArray, uint32_t taskInfoArraySize,
         TaskStats * stats     = findTaskStats(status->xHandle);
 
         // Copy basic info
-        strncpy(info->name, status->pcTaskName, configMAX_TASK_NAME_LEN - 1);
-        info->name[configMAX_TASK_NAME_LEN - 1] = '\0';
-        info->handle                            = status->xHandle;
-        info->state                             = status->eCurrentState;
-        info->priority                          = status->uxCurrentPriority;
-        info->stackHighWaterMark                = status->usStackHighWaterMark;
-        info->runTimeCounter                    = status->ulRunTimeCounter;
-        info->cpuPercentage = totalRunTime > 0 ? (uint32_t) (((uint64_t) status->ulRunTimeCounter * 10000) / totalRunTime) : 0;
+        strncpy(info->stats.name, status->pcTaskName, configMAX_TASK_NAME_LEN - 1);
+        info->stats.name[configMAX_TASK_NAME_LEN - 1] = '\0';
+        info->stats.handle                            = status->xHandle;
+        info->state                                   = status->eCurrentState;
+        info->priority                                = status->uxCurrentPriority;
+
+        // Calculate stack sizes using the stack pointers
+        size_t totalStackSize = (status->pxEndOfStack - status->pxStackBase) * sizeof(StackType_t);
+        size_t usedStackSize  = totalStackSize - (status->usStackHighWaterMark * sizeof(StackType_t));
+
+        info->stackHighWaterMark = usedStackSize;  // Max stack used in bytes
+        info->stackMaxSize       = totalStackSize; // Total allocated stack in bytes
+
+        info->runTimeCounter = status->ulRunTimeCounter;
+        info->cpuPercentage  = totalRunTime > 0 ? (uint32_t) (((uint64_t) status->ulRunTimeCounter * 10000) / totalRunTime) : 0;
 
         // Add tracking stats if available
         if (stats != NULL)
         {
-            info->switchOutCount  = stats->switchOutCount;
-            info->preemptionCount = stats->preemptionCount;
+            info->stats.switchOutCount         = stats->switchOutCount;
+            info->stats.preemptionCount        = stats->preemptionCount;
+            info->lastExecutionTime            = stats->lastSwitchOutTime;
+            info->stats.readyTimeHighWaterMark = stats->readyTimeHighWaterMark;
+            info->stats.totalReadyTime         = stats->totalReadyTime;
+            info->stats.totalRunningTime       = stats->totalRunningTime;
+            info->stats.lastSwitchOutTime      = stats->lastSwitchOutTime;
+            info->stats.lastMovedToReadyTime   = stats->lastMovedToReadyTime;
+            info->stats.lastMovedToRunningTime = stats->lastMovedToRunningTime;
+            info->stats.isDeleted              = stats->isDeleted;
```

Comment:
We should be able to do
info->stats = *stats

Reply:
same, memcpy because of the array no?